### PR TITLE
Fix LoadMcStas crash with single event dataset

### LINKED ISSUE #21610: Mantid sometimes crashes when loading the same mcstas file twice
The zip file:

[nexusMcstasfiles.zip](https://github.com/mantidproject/mantid/files/1651842/nexusMcstasfiles.zip)

contains two files:

* mccode_contains_one_bank.h5
* mccode_multiple_scattering.h5

Mantidplot crashes if mccode_contains_one_bank.h5 is loaded two in a row. This seem to only occur when mcstas date contains just one event dataset. 

In this issue fix this and add:

* unit test that loads mccode_contains_one_bank.h5 twice in a row
* add unit test that loads a mixture of the two files attached
* add mccode_multiple_scattering.h5 to DocTest and use this dataset instead in the LoadMcStas usage example: "Example - Comparing event data entries in a McStas Nexus file"

FI when loading mccode_multiple_scattering.h5 the following workspaces are created:

capture

### PR DESCRIPTION
**Pull request by MatthewBowles**
LoadMcStas would crash when loading certain files due to an out of bounds vector index. This PR fixes the issue, adds tests, and corrects the usage example.

**To test:**

- Run unit tests

Fixes #21610 

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [x] Is the code of an acceptable quality?
- [x] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)?
- [x] Are the unit tests small and test the class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [x] Do changes function as described? Add comments below that describe the tests performed?
- [x] Do the changes handle unexpected situations, e.g. bad input?
- [x] Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/DataHandling/src/LoadMcStas.cpp [hunk 1]
```diff
@@ -152,10 +148,11 @@ void LoadMcStas::exec() {
  * @param outputGroup pointer to the workspace group
  * @param nxFile Reads data from inside first first top entry
  */
-void LoadMcStas::readEventData(
+std::vector<API::IEventWorkspace_sptr> LoadMcStas::readEventData(
```
Comment (Anders-Markvardsen) [inline Framework/DataHandling/src/LoadMcStas.cpp]: Please add code comment to this methods new return value

File: Framework/DataHandling/src/LoadMcStas.cpp [hunk 2]
```diff
@@ -152,10 +148,11 @@ void LoadMcStas::exec() {
...
     const std::map<std::string, std::string> &eventEntries,
     WorkspaceGroup_sptr &outputGroup, ::NeXus::File &nxFile) {
 
+  std::vector<API::IEventWorkspace_sptr> scatteringWS;
```
Comment (Anders-Markvardsen) [inline Framework/DataHandling/src/LoadMcStas.cpp]: Please add code comment explaining scatteringWS

**Comment by Anders-Markvardsen** [comment]
Note it looks like one of the doc tests is failing: http://builds.mantidproject.org/job/pull_requests-ubuntu/19864/

**Comment by MatthewBowles** [comment]
Thanks @Anders-Markvardsen, I think those problems are fixed now.

**Review by Anders-Markvardsen — APPROVED** [review]
Tested manually as well as with the added tests for this alg. Also, good with re-adjustment of code to fix this issue

**Pull request merged**